scmailclient/scmail.py

```diff
@@ -0,0 +1,493 @@
+def export_key(ctx, fingerprint, is_file, is_pvt):
+    file_name = None
+    passphrase = None
+
+    if is_file is True:
+        file_name = click.prompt("Enter the file name", type=str)
+        try:
+            file_name = Path(file_name)
+            file_name.touch(exist_ok=False)
+            logging.debug("file not exists.")
+        except FileExistsError:
+            logging.error("Cause the path already exist, export to file failed.")
+            if click.confirm("Do you want to print on console?", default=True):
+                is_file = False
+            else:
+                logging.warning("Mail Client Exit.")
+                return
+
+            pass
+
+    if is_pvt is True:
+        passphrase = click.prompt("Enter pvt's passphrase", hide_input=True, type=str)
+
+    pub, pvt = ctx.parent.gpg.export_key(
+        fingerprint, export_pvt=is_pvt, password=passphrase
+    )
+
+    if pub == "":
+        logging.error("The key not exists, please check fingerprint.")
+        logging.error("Mail Client Exits.")
+        return
+
+    if is_file is False:
+        ctx.parent.pp.pprint(pub)
+        if is_pvt is True:
+            ctx.parent.pp.pprint(pvt)
+    else:
+        with file_name.open("w") as f:
+            f.write(pub)
+            if pvt:
+                f.write(pvt)
+
+    logging.info("Export key successful.")
+
+
+@click.command(name="import")
+@click.option(
+    "--file-path",
+    "-p",
+    prompt="Enter file path",
+    required=True,
+    type=str,
+    help="The path of imported key.",
+)
+@click.option(
+    "--fingerprint",
+    "-f",
+    prompt="Enter fingerprint",
+    required=True,
+    callback=check_fingerprint,
+    type=str,
+    help="The fingerprint of this key/key pair.",
+)
+@click.pass_context
+def import_key(ctx, file_path, fingerprint):
+    # scaning key
+    res = ctx.parent.gpg.scan_file(file_path)
+
+    if res == []:
+        logging.error(
+            "The file path is wrong. File not found. Or No key (pair) in this file."
+        )
+        return
+
+    if fingerprint != res[0].get("fingerprint"):
+        logging.error(
+            "Unsafe Key.\nThe user input fingerprint is different from computed fingerprint."
+        )
+        logging.error("Mail Client Exits.")
+        return
+
+    # import key
+    try:
+        key_data = Path(file_path).open().read()
+        res = ctx.parent.gpg.import_key(key_data=key_data)
+    except OSError:
+        logging.error("Cannot read the file.")
+        return
+        pass
+
+    logging.info(
+        f"Successful import {res.get('imported')} keys.\nThe fingerprint is:\n{res.get('success')[0].get('fingerprint')}"
+    )
+
+
+client.add_command(create_key)
+client.add_command(list_keys)
+client.add_command(register)
+client.add_command(send)
+client.add_command(retrieve)
+client.add_command(export_key)
+client.add_command(import_key)
+
+
+def main():
+    client(prog_name="scmail")
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Is there any reason this can't just be `client(prog_name="scmail")` and get rid of `main()`?

Reply:
Yes, it can.